### PR TITLE
ADD: SSV operator registered check

### PR DESCRIPTION
**Pull request by PatrickRL**
Checks with the SSV api if the operator has been registered, if so, don't asks them to register anymore and show the dashboard directly.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,5 +3,6 @@
     "source.fixAll.eslint": true
   },
   "editor.formatOnSave": true,
-  "editor.defaultFormatter": "esbenp.prettier-vscode"
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "nuxt.isNuxtApp": false
```
Comment (NeoPlays) [inline .vscode/settings.json:7]: what is this for?

File: launcher/src/components/UI/node-header/ServiceLinks.vue [hunk 2]
```diff
@@ -46,15 +49,34 @@ export default {
...
     ...mapState(useServices, {
       allServices: "allServices",
     }),
+    ...mapWritableState(useNodeHeader, {
+      testOperatorData: "testOperatorData",
+      pubkey: "pubkey",
+    }),
   },
+  mounted() {
+    //console.log(this.pubkey);
```
Comment (NeoPlays) [inline launcher/src/components/UI/node-header/ServiceLinks.vue:66]: this should be deleted

File: launcher/src/components/UI/node-header/ServiceLinks.vue [hunk 3]
```diff
@@ -46,15 +49,34 @@ export default {
...
+  },
+
   methods: {
+    getKeys: async function () {
```
Comment (NeoPlays) [inline launcher/src/components/UI/node-header/ServiceLinks.vue:70]: don't execute these functions in `ServiceLinks.vue` execute them in `SsvModal.vue` where you actually need them then you also don't need the pinia store entries `pubkey` and `testOperatorData`

File: launcher/src/components/UI/node-header/ServiceLinks.vue [hunk 4]
```diff
@@ -86,6 +109,16 @@ export default {
       this.showPrometheusWindow = false;
       this.showMevboostWindow = false;
     },
+    async checkOperator() {
```
Comment (NeoPlays) [inline launcher/src/components/UI/node-header/ServiceLinks.vue:112]: same goes for this function

File: launcher/src/components/UI/services-modal/SsvDashboard.vue [hunk 5]
```diff
@@ -30,12 +30,7 @@ import { useNodeStore } from "@/store/theNode";
 import { useServices } from "@/store/services";
 import ControlService from "@/store/ControlService";
 export default {
-  props: {
-    pubkey: {
-      type: String,
-      required: true,
-    },
-  },
+  props: ["pubkey"],
```
Comment (NeoPlays) [inline launcher/src/components/UI/services-modal/SsvDashboard.vue:33]: don't remove the props typing

File: launcher/src/components/UI/services-modal/SsvModal.vue [hunk 6]
```diff
@@ -81,19 +87,37 @@ export default {
...
+
+    ...mapWritableState(useNodeHeader, {
+      pubkey: "pubkey",
     }),
   },
   mounted() {
     this.getKeys();
   },
+  created() {
+    setTimeout(() => {
```
Comment (NeoPlays) [inline launcher/src/components/UI/services-modal/SsvModal.vue:109]: what if the api call takes longer than 3 seconds? just do it as i stated in the previous comment and execute the functions in this component. and when you use `async await` you can just wait for the api result and make then all the checks

File: launcher/src/store/nodeHeader.js [hunk 7]
```diff
@@ -9,6 +9,8 @@ export const useNodeHeader = defineStore("nodeHeader", {
       isUpdateAvailable: false,
       updating: false,
       operators: [{ operatorName: "stereum" }, { operatorName: "Rocklogic GmbH" }],
+      testOperatorData: "",
```
Comment (NeoPlays) [inline launcher/src/store/nodeHeader.js:12]: you then can remove these 2 lines as well

**Review by NeoPlays — CHANGES_REQUESTED** [review]
pls adjust, dm me if you need more information

File: .vscode/settings.json (hunk 1 above)
Comment (daverolo) [reply .vscode/settings.json:7]: i would recommend to revert it, reza had the same "problem" and also reverted it before merge.

**Review by NeoPlays — APPROVED** [review]
wonderful 🚀

**Pull request merged**